gocheck2/checkers.go

```diff
@@ -138,6 +138,44 @@ func (h *hasKey) Info() *CheckerInfo {
 var HasKey = &hasKey{}
 
 // -----------------------------------------------------------------------
+// NoErr checker.
+
+type noErr struct{}
+
+// This exists to implement fmt.GoStringer and force the `%#v` format to show
+// the string unescaped, newlines and all.
+type rawString string
+
+func (r rawString) GoString() string {
+	return string(r)
+}
+
+func (c noErr) Check(params []interface{}, names []string) (bool, string) {
+	err, ok := params[0].(error)
```

Comment:
is this sufficient to ensure the err is nil?  there's a potential case where the value is nil, but the value's type is error

can you add tests?